src/test/kotlin/test/ktortemplate/core/service/TestCarService.kt

```diff
@@ -0,0 +1,112 @@
+package test.ktortemplate.core.service
+
+import io.ktor.util.KtorExperimentalAPI
+import org.amshove.kluent.`should be equal to`
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.assertThrows
+import org.koin.core.context.startKoin
+import org.koin.core.context.stopKoin
+import org.koin.test.KoinTest
+import org.koin.test.inject
+import org.testcontainers.junit.jupiter.Container
+import org.testcontainers.junit.jupiter.Testcontainers
+import test.ktortemplate.conf.EnvironmentConfigurator
+import test.ktortemplate.containers.PgSQLContainerFactory
+import test.ktortemplate.core.model.CarSaveCommand
+import test.ktortemplate.core.model.Part
+import test.ktortemplate.core.model.RegisterPartReplacementCommand
+import test.ktortemplate.core.persistance.CarRepository
+import test.ktortemplate.core.persistance.PartRepository
+import test.ktortemplate.core.testDatabaseConfigs
+
+@Testcontainers
+@KtorExperimentalAPI
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class TestCarService : KoinTest {
+
+    private val carRepository: CarRepository by inject()
+    private val partRepository: PartRepository by inject()
+    private val carService: CarService by inject()
+
+    companion object {
+        @Container
+        private val dbContainer = PgSQLContainerFactory.instance
+    }
+
+    @BeforeAll
+    fun setup() {
+        val appModules = EnvironmentConfigurator(testDatabaseConfigs).getDependencyInjectionModules()
+        startKoin { modules(appModules) }
```

Comment:
@felix19350  What about if we want to override services in tests, like a mock service for geocoding or storage, we override them here?

Reply:
@pedrohh koin provides a way to override dependencies like so:
```kotlin
startKoin { modules(appModules + module {
    single<CarService>(override = true) { MockedCarService() }
}) }
```

Important note here, these declaration must be always be the last one.

Reply:
@monxalo do you want to add an item to the wiki for that particular piece of knowledge?

Reply:
We might also consider a way to override modules on the environment configuration, as that may be interesting for testing purposes.